sdc/tests/test_series.py

```diff
@@ -3529,6 +3530,82 @@ def test_impl(S1, S2, S3):
         pd.testing.assert_series_equal(hpat_func(S1, S2, S3),
                                        test_impl(S1, S2, S3))
 
+    def sdc_join_series_indexes(self):
+        def test_impl(S1, S2):
+            return S1.add(S2)
+
+        sdc_func = self.jit(test_impl)
+
+        data = [0, 1, 2, 3, 4]
+        index1 = [3.3, 5.4, np.nan, 7.9, np.nan]
+        index2 = [3, 4, 3, 9, 2]
+        S1 = pd.Series(data, index1)
+        S2 = pd.Series(data, index2)
+        pd.testing.assert_series_equal(sdc_func(S1, S2), test_impl(S1, S2))
+
+    # SDC operator methods returns only float Series
+    def test_series_add(self):
+        def test_impl(S1, S2, value):
+            return S1.add(S2, fill_value=value)
+
+        sdc_func = self.jit(test_impl)
+
+        cases_data = [[0, 1, 2, 3, 4], [5, 2, 0, 333, -4], [3.3, 5.4, np.nan, 7.9, np.nan]]
+        cases_index = [[0, 1, 2, 3, 4], [3, 4, 3, 9, 2], None]
+        cases_value = [None, 4, 5.5]
+        for data, index, value in product(cases_data, cases_index, cases_value):
+            with self.subTest(data=data, index=index, value=value):
+                S1 = pd.Series(data, index)
```

Comment:
It seems corruption/compilation failure problems in tests are caused by using index=None as a data in S2 series. I'm still suggesting to not mix up data and indexes to be tested between each other, e.g.:
```
        for value in cases_value:
            for data1, data2 in product(cases_data, cases_data):
                for index1, index2 in product(cases_index, cases_index):
                    S1 = pd.Series(data1, index1)
                    S2 = pd.Series(data2, index2)
                    with self.subTest(left=S1, right=S2, value=value):
...
```